modelbaker/iliwrapper/ilivalidator.py

```diff
@@ -30,10 +30,21 @@
 class Validator(IliExecutable):
     def __init__(self, parent=None):
         super().__init__(parent)
+        self.version = 4
 
     def _create_config(self):
         return ValidateConfiguration()
+        
+    def _get_ili2db_version(self):
+        return self.version
+        
+    def _args(self, hide_password):
+        args = super()._args(hide_password)
 
+        if self.version == 3 and "--export3" in args:
```

Comment:
if it's version 3 then it should remove it (because it does not need it). `--export3` is a version 4 parameter to provide backwards compatibility with 3.

The tool is defined with this functionality. So this is the major definition. 4 is default (soon it will be 5) and if you want to override it you have to set it here (or in the Exporter).

Reply:
Ok I see

Reply:
Now I got it: On validation or export we could have the special case that the tool has not the same version like the db. This means there we should be able to override the tool-version. Usually we use 4 (modelbaker uses 4) but another plugin might want to set it to 3.